File: be/src/main/java/nhom27/itss/be/entity/Notification.java
package nhom27.itss.be.entity;

import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.sql.Timestamp;

@Entity
@Getter
@Setter
@Builder
@NoArgsConstructor
@Table(name = "notifications")
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class Notification {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "notification_id", nullable = false)
    Integer notificationId;


    @Column(name = "message", nullable = false)
    String message;

    @Column(name = "notification_type")
    String notificationType;

    @Column(name = "created_at")
    Timestamp createdAt;

    @Column(name = "is_read")
    Boolean read;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    private User user;


    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "food_id")
    private FoodItem food;

}

File: frontend/User_Mobile_flutter/lib/models/notification.dart
class UserNotification {
  final int notificationId;
  final String message;
  final String notificationType;
  final DateTime createdAt;
  final bool isRead;
  final int userId;
  final int foodId;

  UserNotification({
    required this.notificationId,
    required this.message,
    required this.notificationType,
    required this.createdAt,
    required this.isRead,
    required this.userId,
    required this.foodId,
  });

  factory UserNotification.fromJson(Map<String, dynamic> json) {
    return UserNotification(
      notificationId: json['notificationId'] as int,
      message: json['message'] as String,
      notificationType: json['notificationType'] as String,
      createdAt: DateTime.fromMillisecondsSinceEpoch(json['createdAt']),
      isRead: json['isRead'] as bool,
      userId: json['userId'] as int,
      foodId: json['foodId'] as int,
    );
  }
}
